content/general/limits.textile

```diff
@@ -0,0 +1,314 @@
+
+The limit is calculated based on the hard limit for "peak connections":#peak-connections, with a minimum value of 50 per second for self-service and business accounts.
+
+If the connection rate is exceeded, a "global per second limit":#global-time will be applied to new connection attempts.
+
+h3(#connection-channels). Number of channels
+
+The number of channels per connection are the number of channels each client can be attached to on a "realtime connection":/realtime/connection.
+
+The number of channels per connection is a "quantity-based limit":#quantity-rates. If a connection attempts to exceed the limit on the number of channels it is attached to, the attachment will fail and the error code @90010@ will be returned.
+
+h3(#outbound-message). Outbound message rate
+
+The outbound message rate is the maximum number of messages that can be published per second on a "realtime connection":/realtime/connection.
+
+If the outbound message rate is exceeded for a connection, a "local per second limit":#time will be applied to that connection.
+
+h3(#inbound-message). Inbound message rate
+
+The inbound message rate is the maximum number of messages that can be received per second on a "realtime connection":/realtime/connection.
+
+If the inbound message rate is exceeded for a connection, a "local per second limit":#time will be applied to that connection.
+
+h3(#connection-state). Connection state recovery
+
+Connection state recovery is the duration the Ably service will keep a connection alive for if the client connection drops. This accounts for scenarios such as where a client is switching between WiFi and 4G, or entering a tunnel. If the client reestablishes its connection within the allowed duration, all messages and presence events will be replayed to that client in order.
+
+If a connection isn't reestablished in time, the client is detached from its channels and the error code @80008@ is returned.
+
+h2(#channel). Channel limits
+
+Channel limits relate to the number, rate and membership of "channels":/glossary#channels on your account.
+
+|_=. Limit |_=. Free |_=. Self-service |_=. Business |_=. Enterprise |
+| "Number of subscribers":#subscribers (per channel) |\4=. Unlimited |
+| "Presence members":#presence-members (per channel) | 50 | 200 | 200 | Custom |
+| "Channel creation rate":#channel-create (per second) | 20 | >=50 | >=50 | Custom |
+| "Message publish rate":#message-publish-rate (per second) | 15 | 50 | 50 | Custom |
+
+h3(#subscribers). Number of subscribers
+
+There is no limit on the number of clients that can be subscribed to a channel.
+
+h3(#presence-members). Presence members
+
+The number of clients that can use the "presence feature":/realtime/presence in a channel is limited as a precaution. If a large number of clients are subscribed to presence events on a channel then the amount of messages can scale very quickly.
```

Comment:
"limited as a precaution" - makes it sound like we're unsure what will happen.

The number of clients that can use the "presence feature" in a channel is limited - I think this is imprecise. The number of clients that can be simultaneously present in a given channel is limited.

I think this simply needs to say:

The number of clients that can be simultaneously "present":/realtime/presence in a given channel is limited.

The presence member limit exists to ensure that the size of the aggregate state associated with a channel is limited, but also that the aggregate rate for presence messages is supportable. It is common for the present members on a channel to behave similarly, for example all joining or leaving at a similar time. As an example, ....

Reply:
I agree on this point - I should have expanded on the "precaution" part if going that route. I've removed it and reworded the section based on your feedback.